### PR TITLE
Match identity key specs

### PR DESCRIPTION
**Pull request by devceline**
# Changes
Make identity keys use domain of external dapp and accept a statement. 

Match the following specs: https://github.com/WalletConnect/walletconnect-specs/pull/138 and https://github.com/WalletConnect/walletconnect-specs/pull/139

### CODE REVIEW

**Review by bkrem — APPROVED** [review]
LGTM, since this is a breaking change/major version bump, we should really wait for specs to merge before releasing this on `latest`.

Else we may end up rebumping major version here if something else changes in spec last minute.

Can of course already use this via canaries in the meantime if necessary 👍

**Review by Cali93 — APPROVED** [review]
LGTM👌

File: misc/identity-keys/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@walletconnect/identity-keys",
-  "version": "0.1.10",
+  "version": "0.12.0-8d867",
```
Comment (Cali93) [inline misc/identity-keys/package.json:3]: Looks like we jumped a few versions here

**Review by Cali93 — CHANGES_REQUESTED** [review]

File: misc/identity-keys/package.json (hunk 1 above)
Comment (devceline) [reply misc/identity-keys/package.json:3]: that is weird, I'll fix that up

**Review by Cali93 — APPROVED** [review]

**Pull request merged**